test/migration-upgrade.test.ts

```diff
@@ -0,0 +1,267 @@
+import * as hre from "hardhat";
+import { impersonateAccount } from "@nomicfoundation/hardhat-network-helpers";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import {
+  MockToken,
+  MockToken__factory,
+  ZStakeCorePool,
+  ZStakeCorePool__factory,
+  ZStakeCorePoolMigration, ZStakePoolFactory__factory
+} from "../typechain";
+import { expect } from "chai";
+import { compareStorageData, ContractStorageData, readContractStorage } from "../src/migration/storage-data";
+import { upgradeStakingPool } from "../src/migration/upgrade-pool";
+import { transferProxyAdminOwnership } from "../src/migration/transfer-ownership";
+import {
+  FACTORY_ADDRESS,
+  LP_POOL_ADDRESS, LP_TOKEN_ADDRESS,
+  OWNER_SAFE_ADDRESS,
+  WILD_POOL_ADDRESS,
+  WILD_TOKEN_ADDRESS
+} from "../src/migration/constants";
+
+
+describe.only("Migration Upgrade", () => {
+  let owner : SignerWithAddress;
+  let tokenVault : SignerWithAddress;
+  let upgradeOwner : SignerWithAddress;
+
+  let poolContractFactory : ZStakeCorePool__factory;
+
+  let wildPool : ZStakeCorePool;
+  let lpPool : ZStakeCorePool;
+  let wildToken : MockToken;
+  let lpToken : MockToken;
+
+  // upgraded
+  let wildPoolNew : ZStakeCorePoolMigration;
+  let lpPoolNew : ZStakeCorePoolMigration;
+
+  let lpPoolStatePreUpgrade : ContractStorageData;
+  let wildPoolStatePreUpgrade : ContractStorageData;
+
+  let originalConsoleLog : typeof console.log;
+
+  before(async () => {
+    originalConsoleLog = console.log;
+    // disable console logging for tests
+    console.log = () => {};
```

Comment:
why do we do this just to reset it in the `afterEach` hook? I don't see either being used in the code